### DISCUSSION TITLE
Loading variants of a robot description

### DISCUSSION THREAD

**Discussion by stephane-caron** [robot-descriptions/robot_descriptions.py#102] [Ideas]
Some descriptions have variants (like the "left" and "right" versions of the Allegro hand) that are currently only available when importing submodules. We could add a ``variant`` keyword argument to `load_robot_description` so that users can access it from there as well.

**Comment by kevinzakka** [comment]
Also useful for loading end-effector-less versions of a robot (e.g. Panda, xArm7) which ship with an end-effector.

**Comment by lvjonok** [comment]
I'd like to propose another feature request. Since Pinocchio `v3.0.0`, it's possible to use MJCF to create models. However, the Pinocchio loader in `robot_descriptions` currently utilizes the `buildFromURDF` method.

Do you think this could be better addressed using a `variant` keyword, or should we handle it with an `if-statement` to check whether `module.MJCF_PATH` or `module.URDF_PATH` is available?

**Reply by stephane-caron** [reply]
Thanks for proposing this. I'd say we separate variants (e.g. left-handed, right-handed, with/without end-effector, etc., of a given robot) from formats (URDF/MJCF) for now. We can then go for your second proposal https://github.com/robot-descriptions/robot_descriptions.py/pull/101.

**Comment by stephane-caron** [comment]
Following up in this thread to note that https://github.com/robot-descriptions/robot_descriptions.py/pull/76 added loading variants for MuJoCo descriptions.
